File: dynamic_listings/lua/default/collisiondetection/line_circle_partial.lua
Point = {}

function Point:new(o)
    -- This is an example constructor
    o = o or {x=0, y=0}
    setmetatable(o, self)
    self.__index = self
    return o
end

Line = {}

function Line:new(o)
    -- This is an example constructor
    o = o or {A=Point:new(), B=Point:new()}
    setmetatable(o, self)
    self.__index = self
    return o
end

Circle = {}

function Circle:new(o)
    -- This is an example constructor
    o = o or {center=Point:new(), radius=0}
    setmetatable(o, self)
    self.__index = self
    return o
end

-- ...

local function line_circle_collision(circle, line)
    local collides_A = circle_point_collision(circle, line.A)
    local collides_B = circle_point_collision(circle, line.B)
    if (collides_A or collides_B) then
        return true
    end
    -- ...
end

File: dynamic_listings/javascript/default/collisiondetection/line_circle_partial.js
class Point{
    // Rewritten as a memo
    constructor(){
        this.x = null;
        this.y = null;
    }
}

class Line{
    constructor(){
        this.A = new Point();
        this.B = new Point();
    }
}

class Circle{
    // Let's define a circle class/structure
    constructor(){
        this.center = new Point();
        this.radius = 0;
    }
}

// ...

function line_circle_collision(circle, line){
    let collides_A = circle_point_collision(circle, line.A);
    let collides_B = circle_point_collision(circle, line.B);
    if (collides_A || collides_B){
        return true;
    }
    // ...
}
